### DISCUSSION TITLE
DataTable Sum over filtered Column

### DISCUSSION THREAD

**Discussion by hschneider** [primefaces/community#2841] [PrimeReact]
Can someone give a simple example., how to  build the total sum over a filtered DataTable column? So that the sum changes with each filter action?

**Comment by qwadrox** [comment]
@hschneider Hi!

Here is a basic example: https://stackblitz.com/edit/udsgcw?file=src%2Fapp%2Ftable-template-demo.html
I used Javascript reduce to sum the prices:

```ts
onFilter(event: TableFilterEvent) {
  const filteredProducts = event.filteredValue;
  this.totalPrice = filteredProducts.reduce(
    (sum, product) => sum + product.price,
    0
  );
}
```
If you need to walk through an array and "reduce" it to a single value, the handy way is to use JS reduce.

**Comment by hschneider** [comment]
THX a lot for pointing the right direction :-)

**Reply by qwadrox** [reply]
@hschneider Hey. I accidentally mixed up the discussions and I've just realized you asked the question for PrimeReact, not PrimeNg.
Here is the PrimeReact example: https://stackblitz.com/edit/tsniev?file=src%2FApp.jsx

I used [onValueChange](https://primereact.org/datatable/#api.DataTable.callbacks.onValueChange) but the idea is identical. 
```ts
<DataTable
    onValueChange={(filteredData) => calculateTotalBalance(filteredData)}
>

const calculateTotalBalance = (data) => {
  const sum = data.reduce((acc, customer) => acc + customer.balance, 0);
  setTotalBalance(sum);
};
```
